freqtrade/persistence/models.py

```diff
@@ -504,14 +513,49 @@ def update_trade(self, order: Order) -> None:
             raise ValueError(f'Unknown order type: {order.order_type}')
         Trade.commit()
 
+    def process_sell_sub_trade(self, order: Order, is_closed: bool = True,
+                               is_non_bt: bool = True) -> None:
+        sell_amount = order.safe_amount_after_fee
+        sell_rate = order.safe_price
+        sell_stake_amount = sell_rate * sell_amount * (1 - self.fee_close)
+        if sell_amount == self.amount:
+            if is_closed:
+                self.close(sell_rate)
+                if is_non_bt:
+                    Trade.commit()
```

Comment:
This seems unnecessarily complex. 
if you remove all commits from this method, and don't return after `process_sell_sub_trade()` in `update_trade()` you'll get the same result - without the need to explicitly differentiate between bt and non-bt - allowing you to remove the `is_non_bt` parameter.

Reply:
updated

Reply:
Please also remove the unnecessary parameter `is_non_bt` from the function header.

Reply:
updated